mne/io/proj.py

```diff
@@ -108,7 +108,10 @@ def add_proj(self, projs, remove_existing=False, verbose=None):
     def add_eeg_average_proj(self):
         """Add an average EEG reference projector if one does not exist
         """
-        if _needs_eeg_average_ref_proj(self.info):
+        if self.info['custom_ref_applied']:
+            logger.warning('A custom reference was previously applied, '
+                           'average reference cannot be created.')
```

Comment:
> It is perfectly valid to move from a custom referencing scheme to an average reference. There are even valid use cases for this.

I think you might have a point that lines 111-113 might be too restrictive, so I'll reverse the change. If the data were acquired with a single reference then yes: we can apply whatever funky multi-channel off-line reference we want, then calculate the CAR and all is well. 

But when we "inject" a Ref-channel(s), the channel count goes up by one (or more), which to my mind is like "inventing" a DOF the data doesn't have. What are the consequences to a possible subsequent inverse operator, if any?